packages/core/src/types/IKeyManager.ts

```diff
@@ -214,6 +219,9 @@ export interface IKeyManager extends IPluginMethodMap {
    * Returns an existing key
    */
   keyManagerGet(args: IKeyManagerGetArgs): Promise<IKey>
+  keyManagerGetWhere(args: IKeyManagerGetWhereArgs): Promise<ManagedKeyInfo>
```

Comment:
I'm surprised we didn't already have a function like this.

We definitely need a way to find keys by
* type
* kms
* the DIDs they are associated with
* algorithms
* metadata matches (this implies unstructured JSON queries)